update langchain agent with qwen2.5 for better accuracy

openvino-dev-samples opened the pull request:
![image](https://github.com/user-attachments/assets/b656adab-a448-4784-a6df-a068e0cb45bb)

openvino-dev-samples committed:
update langchain agent with qwen2.5 for better accuracy

openvino-dev-samples committed:
fix ci issues

fix ci issues

eaidova commented:
@openvino-dev-samples please update test patching:
```
  File "/tmp/scripts/patch_notebooks.py", line 204, in <module>
    patch_notebooks(args.notebooks_dir, args.test_device, args.skip_ov_install)
  File "/tmp/scripts/patch_notebooks.py", line 181, in patch_notebooks
    raise ValueError(f"Processing {notebookfile} failed: {source_value} does not exist in cell")
ValueError: Processing /tmp/notebooks/llm-agent-react/llm-agent-react-langchain.ipynb failed: mistralai/Mistral-7B-Instruct-v0.3 does not exist in cell
```

eaidova commented on notebooks/llm-agent-react/llm-agent-react-langchain.ipynb:
```diff
@@ -279,22 +320,57 @@
    "metadata": {
     "test_replace": {
-     "mistralai/Mistral-7B-Instruct-v0.3": "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+     "Qwen/Qwen2.5-7B-Instruct": "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
```
Sometimes Qwen2.5-3B can not provide good results

eaidova replied:
```suggestion
     "Qwen/Qwen2.5-7B-Instruct": "Qwen/Qwen2.5-1.5B-Instruct"
```

eaidova committed:
Update notebooks/llm-agent-react/llm-agent-react-langchain.ipynb

openvino-dev-samples commented:
I think a small model may lead parsing error on output of agent.

eaidova commented:
>I think a small model may lead parsing error on output of agent.


@openvino-dev-samples ok, then I think it will be better to remove patching and disable notebook for precommit as we can not run 7B models due to github limitations (it is still will validated in internal infra with more powerful hw)

openvino-dev-samples commented:
> > I think a small model may lead parsing error on output of agent.
> 
> @openvino-dev-samples ok, then I think it will be better to remove patching and disable notebook for precommit as we can not run 7B models due to github limitations (it is still will validated in internal infra with more powerful hw)

I'dont know if "TinyLlama/TinyLlama-1.1B-Chat-v1.0" can pass the CI, let me try

openvino-dev-samples committed:
rebase

eaidova commented:
> > > I think a small model may lead parsing error on output of agent.
> > 
> > 
> > @openvino-dev-samples ok, then I think it will be better to remove patching and disable notebook for precommit as we can not run 7B models due to github limitations (it is still will validated in internal infra with more powerful hw)
> 
> I'dont know if "TinyLlama/TinyLlama-1.1B-Chat-v1.0" can pass the CI, let me try

what about Qwen2.5-1.5B-Instruct? do you think it less accurate than tinyllama?

openvino-dev-samples commented:
> > > > I think a small model may lead parsing error on output of agent.
> > > 
> > > 
> > > @openvino-dev-samples ok, then I think it will be better to remove patching and disable notebook for precommit as we can not run 7B models due to github limitations (it is still will validated in internal infra with more powerful hw)
> > 
> > 
> > I'dont know if "TinyLlama/TinyLlama-1.1B-Chat-v1.0" can pass the CI, let me try
> 
> what about Qwen2.5-1.5B-Instruct? do you think it less accurate than tinyllama?

It depends on how good LLM can follow prompt's instruction, Let me try ting-llama first

openvino-dev-samples committed:
switch test model

replace llm test case

replace llm test case

replace llm test case

replace llm test case

openvino-dev-samples committed:
replace optimum-cli method

replace optimum-cli method

openvino-dev-samples committed:
replace optimum-cli method

openvino-dev-samples committed:
add langchain-community

openvino-dev-samples committed:
ignore langchain agent notebook
